libs/prelude/Prelude/Strings.idr

```diff
@@ -276,6 +266,34 @@ unwords' ws = assert_total (foldr1 addSpace ws)
 unwords : List String -> String
 unwords = pack . unwords' . map unpack
 
+||| Joins the character lists by newlines into a single character list.
+|||
+||| ```idris example
+||| unlines' [['l','i','n','e'], ['l','i','n','e','2'], ['l','n','3'], ['D']]
+||| ```
+unlines' : List (List Char) -> List Char
+unlines' [] = []
+unlines' ws = assert_total (foldr1 addLine ws) where
+  addLine : List Char -> List Char -> List Char
+  addLine w s = w ++ ('\n' :: s)
```

Comment:
Just a nit, but `w` in unwords presumably stands for "word", so `l` would be more logical here.